### DISCUSSION TITLE
Recommendations for MQTT software?

### DISCUSSION THREAD

**Discussion by Troubadix** [tbnobody/OpenDTU#2001] [General]
Hi,

I'm looking for a easy to install, easy to configure, easy to use MQTT software running on Raspberry Pi OS providing statistics for OpenDTU.

- Receive values from OpenDTU via MQTT and store it in a database (MariaDB, flatfile, ...)
- show current values P, A, U (watts, ampere, volts)
- show statistics (numbers and graphs) of last hours/today/yesterday/last month/13th Nov 2023/last year/...

Any recommendations?
Thanks

**Comment by John-Corey** [comment]
I am using these products for MQTT on a RasPi 3B:
MQTT: Mosquitto
DB: InfluxDB
Control: NodeRED
Visualisation: Grafana

It's a very common combination of tools used by many people (maybe beside the oldfashioned 3B, but I had it already and it works fine from SSD too). I would not say it's easy to get it running, but there are a many tutorials available that will help you. I spent 5 nights until it was running and I no experience with these tools before - so it is not rocket-science and absolutely worth to invest the time as you can customize your setup late to your preferences easily.

HTH
